### DISCUSSION TITLE
ELRS version detection via LUA

### DISCUSSION THREAD

**Discussion by alexey-gamov** [ExpressLRS/ExpressLRS#2368] [Q&A] [answered]
Sup! I maintain a telemetry script for old monochrome transmitters. So far i have a problem with displaying selected receiver output frequency via FMD telemetry channel.

So different ELRS firmwares has different FMD channels count and values.

Is there way to detect what firmware version is used to show appropriate freq on the screen?

May be it's good to get transceiver module version and do this magic..

Need your help community!

**Comment by alexey-gamov** [comment, marked as answer]
thx @CapnBry 🤝🏻😅

```local function parseDeviceInfo(data)
  if data[2] ~= 0xEE then return end -- only interested in TX info
  local name, off = fieldGetString(data, 3)
  mod.name = name
  -- off = serNo ('ELRS') off+4 = hwVer off+8 = swVer
  mod.vMaj = data[off+9]
  mod.vMin = data[off+10]
  mod.vRev = data[off+11]
  mod.vStr = string.format("%s (%d.%d.%d)",
    mod.name, mod.vMaj, mod.vMin, mod.vRev)
  if mod.vMaj == 3 then
    mod.RFMOD = {"", "25Hz", "50Hz", "100Hz", "100HzFull", "150Hz", "200Hz", "250Hz", "333HzFull", "500Hz", "D250", "D500", "F500", "F1000" }
   -- Note: Always use 2.4 limits
    mod.RFRSSI = {-128, -123, -115, -117, -112, -112, -112, -108, -105, -105, -104, -104, -104, -104}
  else
    mod.RFMOD = {"", "25Hz", "50Hz", "100Hz", "150Hz", "200Hz", "250Hz", "500Hz"}
    mod.RFRSSI = {-128, -123, -115, -117, -112, -112, -108, -105}
  end
  return true
end```

**Comment by CapnBry** [comment]
Glad you found it! However, you must me 100% certain, CERTAIN, you only retrieve the information from the module once. Once you have it, be sure to not request the info again. Requesting the info seriously degrades the performance of the RC link temporarily as the TX module queries all connected devices for information, so once you have the info, make certain you do not keep requesting it.
